src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -154,33 +154,12 @@ private static ImmutableArray<BoundExpression> CollectBinaryOperatorInterpolated
             Debug.Assert(node.OperatorKind == BinaryOperatorKind.StringConcatenation);
             Debug.Assert(node.InterpolatedStringHandlerData is not null);
             var partsBuilder = ArrayBuilder<BoundExpression>.GetInstance();
-            while (true)
+            node.VisitBinaryOperatorInterpolatedString(partsBuilder, (BoundInterpolatedString interpolatedString, ArrayBuilder<BoundExpression> partsBuilder) =>
```

Comment:
I'm really liking the new helper methods to abstract the traversal. It much simplifies the different places that need to visit. Thanks for making that change